src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHelpers.cs

```diff
@@ -33,15 +36,32 @@ private static bool CheckParentIsSystemd()
             try
             {
                 // Check whether our direct parent is 'systemd'.
+                // This will not work for containerised applications, since our direct parent will be some container runtime
                 int parentPid = Interop.libc.GetParentPid();
                 string ppidString = parentPid.ToString(NumberFormatInfo.InvariantInfo);
                 byte[] comm = File.ReadAllBytes("/proc/" + ppidString + "/comm");
-                return comm.AsSpan().SequenceEqual(Encoding.ASCII.GetBytes("systemd\n"));
+                if (comm.AsSpan().SequenceEqual(Encoding.ASCII.GetBytes("systemd\n")))
```

Comment:
After moving the PID 1 check before this, this can go back to what it was.

Reply:
Actioned